openedx/features/journals/api.py

```diff
@@ -0,0 +1,276 @@
+        """ Return journals worker """
+        return User.objects.get(username=JOURNAL_WORKER_USERNAME)
+
+
+def fetch_journal_access(site, user):   # pylint: disable=unused-argument
+    """
+    Retrieve journal access record for given user.
+    Retrieve if from the cache if present, otherwise send GET request to the journal access api
+        and store it in the cache
+
+    Args:
+        site (Site)
+        user (username | str): user to retrieve access records for
+
+    Returns:
+        list of dicts: list of journal access dicts
+
+    Raises:
+        ConnectionError: raised if lms is unable to connect to the journals service.
+        SlumberBaseException: raised if API response contains http error status like 4xx, 5xx etc...
+        Timeout: raised if API is talking to long to respond
+    """
+    # TODO: WL-1560: (see jira for more info)
+    journal_access_records = JournalsApiClient().client.journalaccess.get(
+        user=user,
+        get_latest=True
+    )
+
+    return journal_access_records.get('results', [])
+
+
+def get_cache_key(**kwargs):
+    """
+    Get MD5 encoded cache key for given arguments.
+
+    Here is the format of key before MD5 encryption.
+        key1:value1__key2:value2 ...
+
+    Example:
+        >>> get_cache_key(site_domain="example.com", resource="enterprise-learner")
+        # Here is key format for above call
+        # "site_domain:example.com__resource:enterprise-learner"
+        a54349175618ff1659dee0978e3149ca
+
+    Arguments:
+        **kwargs: Key word arguments that need to be present in cache key.
+
+    Returns:
+         An MD5 encoded key uniquely identified by the key word arguments.
+    """
+    key = '__'.join(['{}:{}'.format(item, value) for item, value in six.iteritems(kwargs)])
+
+    return hashlib.md5(key).hexdigest()
+
+
+def journals_enabled():
+    """
+    Determines whether the Journals app is installed and enabled
+    """
+    return 'openedx.features.journals.apps.JournalsConfig' \
+        in settings.INSTALLED_APPS and settings.FEATURES.get('ENABLE_JOURNAL_INTEGRATION', False)
+
+
+def get_journals(site):
+    """Retrieve journals from the discovery service.
+
+    Keyword Arguments:
+        site (Site): Site object for the request
+        will be returned
+
+    Returns:
+        list of dict, representing journals
+    """
+    if not journals_enabled():
+        return []
+
+    api_resource = 'journals'
+    orgs = configuration_helpers.get_current_site_orgs()
+
+    cache_key = get_cache_key(
+        site_domain=site.domain,
+        resource=api_resource,
+        orgs=orgs
+    )
+
+    # look up in cache
+    journals = cache.get(cache_key)
+
+    if not journals:
+        api_client = DiscoveryApiClient()
+        if not api_client:
+            return []
+        journals = api_client.get_journals(orgs)
+        cache.set(cache_key, journals, JOURNALS_CACHE_TIMEOUT)
+
+    return journals
+
+
+def fix_course_images(bundle):
```

Comment:
Rather post-fixing, can we have the API endpoint return the right images and just depend on that?

Reply:
@tuchfarber 
please address

Reply:
The endpoint in discovery is just returning courses using the standard serializer. The two options would be to create a new serializer based off of the standard one, or edit the standard one. I assume we definitely don't want to edit the standard one, and I didn't think we'd want a course serializer in the Journals app.